docs/db-migration-guide.ipynb

```diff
@@ -0,0 +1,137 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Charmed Kubeflow Databases Migration Guide\n",
+    "\n",
+    "There are two databases in Charmed Kubeflow - `katib-db` and `kfp-db`. In Charmed Kubeflow 1.7 these MariaDB databases deployed by `charmed-osm-mariadb-k8s` charm. Since release 1.7 there is an option to relate charm to new MySQL database that is deployed by `mysql-k8s` charm. In cases where previous deployment has MariaDB a migration of database is required."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Prerequisites\n",
+    "\n",
+    "In order to perform migaraion the following tools should be installed on client machine that has access to Charmed Kubeflow deployment:\n",
+    "- `mysql-client` (install by running `sudo apt install mysql-client`)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Katib DB migration"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "vscode": {
+     "languageId": "shellscript"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# obtain username and password of existing MariadDB database from DB relation\n",
+    "# in Charmed Kubeflow 1.6 and 1.7 username used is `root` and password is specified in `mysql` relation by\n",
+    "# 'root_password'\n",
+    "KATIB_DB_MYSQL_RELATION_ID=$(juju show-unit katib-db/0 | yq '.[] | .relation-info | select(.[].endpoint == \"mysql\") | .[0] | .relation-id')\n",
+    "KATIB_DB_USER=root\n",
+    "KATIB_DB_PASSWORD=$(bash -c \"juju run --unit katib-db/0 'relation-get -r $KATIB_DB_MYSQL_RELATION_ID - katib-db/0' | grep root_password\" | awk '{print $2}')\n",
```

Comment:
I had an issue with this command:

```
KATIB_DB_PASSWORD=$(bash -c "juju run --unit katib-db/0 'relation-get -r $KATIB_DB_MYSQL_RELATION_ID - katib-db/0' | grep root_password" | awk '{print $2}')
ERROR option provided but not defined: --unit
```

Reply:
I removed `--unit`, but now I get

```
KATIB_DB_PASSWORD=$(bash -c "juju run katib-db/0 'relation-get -r $KATIB_DB_MYSQL_RELATION_ID - katib-db/0' | grep root_password" | awk '{print $2}')
ERROR invalid unit or action name "relation-get -r 14 - katib-db/0"
```

I am using juju 3.1/stable, though

Reply:
Strange. Need to verify.
But guide target migration from MariaDB, which won't run on juju 3.1. Do we want to specify what version of Juju should be used?
Also, do you think it is worth mentioning that first we do DB migration then we do Juju migration? Or should we leave it out of scope for this?

Reply:
Added Juju 2.9 as requirement. Command should work.

Reply:
I think that, since we are going to start supporting 3.1 soon, we should mention how to deal with both cases. Also, 1.8 only supports 3.1, which will require some instructions for what to do first. Should I upgrade my cluster first, then juju then do the DB migration?

Reply:
We have it in Jira, I will add it here: DB migration must be done prior to any other upgrade and it is all done useing Juju 2.9.

This makes mentioning of 3.1 out of scope for this guide. This should be left for overlall CKF upgrade guide.